datatypes/src/collections/feature_collection.rs

```diff
@@ -991,12 +991,12 @@ where
                 }
                 FeatureDataType::Text => {
                     let array: &arrow::array::StringArray = downcast_array(column);
-                    TextDataRef::new(
-                        array.value_data(),
-                        array.value_offsets(),
-                        array.data_ref().null_bitmap(),
-                    )
-                    .into()
+                    let fixed_nulls = if column.null_count() > 0 {
```

Comment:
This is cheap to call, right?

Reply:
Yes. The null values are counted at some point during creation of the column in every case. This function only fetches the pre-calculated number.